File: backend/src/main/java/com/example/backend/model/Order.java
package com.example.backend.model;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Entity
@Table
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Order {
        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        private Long id;
        private String receiverAddress;
        private int extraContactNumber;
        @OneToMany(mappedBy = "order", fetch = FetchType.LAZY )
        private List<OrderDetail> orderDetailList;


        public Order(Long id) {
                this.id = id;
        }

}

File: frontend/src/app/dto/Order.ts
import { OrderDetail } from './OrderDetail';
import { User } from './User';

export class Order {

  id: number | null = null;
  receiver: User | null = null;
  receiverAddress: string | null = null;
  extraContactNumber: number | null = null;
  orderedBy: User | null = null;

  orderDetailList: Array<OrderDetail> = [];

  constructor(o?: Partial<Order>) {
    Object.assign(this, o);
  }
}
